.github/workflows/build-shared-library.yml

```
@@ -24,5 +24,4 @@ jobs:
         run: ./build/main-psar
 
       - name: run gtest
-        run: timeout 10s ./build/test-psar --output-on-failure
-
+        run: timeout 20s make test
```

Je pense que ajouter valgrind est une très bonne idée. 
```suggestion
        run: timeout 2m valgrind --leak-check=full make test
```

---

Oui, ce serait pas mal pour les memory leak
C'est cool de régler ça dès maintenant plutôt que plus tard
Après dans les tests je n'ai pas libéré les mémoires
Je quittais juste le programme

---

Ha bas non faut libérer la mémoire mdr. Sinon on ne sait pas si ça fonctionne vraiment.

---

Yipp, vais ajouter les instructions pour free la mémoire allouée

---

après ce qui est cool, c'est que les donnés sont still reachable, donc pas de memory leak pour le moment
et petit soucis, il arrive pas à détecter valgrind ici...

---

Il faut l'installer avec apt install valgrind avant de l'utiliser.

---

tu peux l'ajouter plss

---

C'est bon, je l'ai ajouté.